agent/consul/acl_test.go

```diff
@@ -565,13 +611,19 @@ func TestACLResolver_DownPolicy(t *testing.T) {
 		require.True(t, authz == authz2)
 		require.True(t, authz.NodeWrite("foo", nil))
 
+		requirePolicyCached(t, r, "node-wr", true, "cached")    // from "found" token
+		requirePolicyCached(t, r, "dc2-key-wr", true, "cached") // from "found" token
+
 		// the go routine spawned will eventually return with a authz that doesn't have the policy
 		retry.Run(t, func(t *retry.R) {
 			authz3, err := r.ResolveToken("found")
 			assert.NoError(t, err)
 			assert.NotNil(t, authz3)
 			assert.False(t, authz3.NodeWrite("foo", nil))
 		})
+
+		requirePolicyCached(t, r, "node-wr", false, "no longer cached")    // from "found" token
```

Comment:
I think this the expected behavior in the async examples, but a second opinion would be nice.